src/pages/search.tsx

```diff
@@ -1,67 +1,157 @@
-import { SEARCH_SKILLS_AND_PROFILES_QUERY } from "../graphql/queries/skills";
+import {
+  GetUserDesireOnEachSkillQuery,
+  SearchSkillsAndProfilesQuery,
+  Skill,
+} from "../generated/graphql";
+import {
+  GET_USER_DESIRE_ON_EACH_SKILL,
+  SEARCH_SKILLS_AND_PROFILES_QUERY,
+} from "../graphql/queries/skills";
 import { i18nContext } from "../utils/i18nContext";
+import CustomSelect from "../components/CustomSelect";
 
 const Search = ({ pathName }) => {
-  const { query } = useRouter();
+  const { t } = useContext(i18nContext);
 
   const isDesktop = useMediaQuery({
     query: "(min-device-width: 1280px)",
   });
-  const { t } = useContext(i18nContext);
+
+  const choicesSorter = [
+    { label: "trendsAsc", value: `${t("search.trends")}` },
+    { label: "default", value: `${t("search.alphabetical")}` },
+    { label: "mostNoted", value: `${t("search.mostNoted")}` },
+  ];
+
   const [search, setSearch] = useState("");
+  const [filter, setFilter] = useState(choicesSorter[0]);
 
   const { data, error } = useQuery<SearchSkillsAndProfilesQuery>(
     SEARCH_SKILLS_AND_PROFILES_QUERY,
     {
       variables: { search: `%${search}%` },
     }
   );
-  if (error) {
-    console.error(error);
-  }
+
+  const { data: responseSkillsDetails, error: skillsDetailsError } =
+    useQuery<GetUserDesireOnEachSkillQuery>(GET_USER_DESIRE_ON_EACH_SKILL, {
+      variables: { search: `%${search}%` },
+    });
+
+  if (error) console.error(error);
+
+  if (skillsDetailsError) console.error(skillsDetailsError);
+
   const skills = data?.skills;
+  const skillsDetails = responseSkillsDetails?.Skill;
   const profiles = data?.profiles;
 
+  const sortedSkills = () => {
+    if (skills && skillsDetails) {
+      const xSkills = [...skills];
+      const skillsDetailsSorted = [...skillsDetails];
+      switch (filter.label) {
+        case "default":
+          if (search === "") {
+            return xSkills.slice(0, 10);
+          }
+          return skills;
+        case "trendsAsc":
+          if (search === "") {
+            return xSkills
+              .sort(
+                (a, b) =>
+                  Number(b.desireLevel + b.skillLevel) / 2 -
+                  Number(a.desireLevel + a.skillLevel) / 2
+              )
+              .slice(0, 10);
+          }
+          return xSkills.sort(
+            (a, b) =>
+              Number(b.desireLevel + b.skillLevel) / 2 -
+              Number(a.desireLevel + a.skillLevel) / 2
+          );
+        case "mostNoted":
+          if (search === "") {
+            return skillsDetailsSorted
+              .sort(
+                (a, b) => b.UserSkillDesires.length - a.UserSkillDesires.length
+              )
+              .slice(0, 10);
+          }
+          return skillsDetailsSorted.sort(
+            (a, b) => b.UserSkillDesires.length - a.UserSkillDesires.length
+          );
+        default:
+          return skills;
+      }
+    } else {
+      return [""];
```

Comment:
```suggestion
      return [];
```

Otherwise, it's showing a SkillPanel component with nothing during the loading